src/librustc_mir/interpret/eval_context.rs

```diff
@@ -600,35 +600,30 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             extra,
         });
 
-        // don't allocate at all for trivial constants
-        if body.local_decls.len() > 1 {
```

Comment:
We need a perf run to see if losing this optimization is a problem.

Reply:
Yeah. It seems like I broke const prop (and other things?), so that should be fixed first.

Reply:
The failure looks like you just need to re-bless `mir-opt`?